lib/sorbet-rails/routes_rbi_formatter.rb

```diff
@@ -4,62 +4,72 @@ class SorbetRails::RoutesRbiFormatter
 
   sig { void }
   def initialize
+    @parlour = T.let(Parlour::RbiGenerator.new, Parlour::RbiGenerator)
     @buffer = T.let([], T::Array[T.any(String, T::Array[String])])
   end
 
   sig { params(title: String).void }
   def section_title(title)
-    @buffer << "\n# Section #{title}"
+    nil
   end
 
   sig { params(routes: T::Array[T::Hash[Symbol, String]]).void }
   def section(routes)
-    @buffer << draw_section(routes)
+    draw_section(routes)
   end
 
-  sig { params(routes: T.untyped).returns(NilClass) }
+  sig { params(routes: T.untyped).void }
   def header(routes)
+    @parlour.root.add_comments([
+      'This is an autogenerated file for Rails routes.',
+      'Please run rake rails_rbi:routes to regenerate.'
+    ])
+
+    @parlour.root.create_class('ActionController::Base') do |klass|
+      klass.create_include('GeneratedUrlHelper')
+    end
   end
 
   sig { params(routes: T.untyped, filter: T.untyped).void }
-  def no_routes(routes=nil, filter=nil)
-    @buffer <<
-      <<~MESSAGE
-        # You do not have any routes defined!
-        # Please add some routes in config/routes.rb.
-        # For more information about routes, see the Rails guide: https://guides.rubyonrails.org/routing.html.
-      MESSAGE
+  def no_routes(routes = nil, filter = nil)
+    @parlour.root.add_comments([
+      'You do not have any routes defined!',
+      'Please add some routes in config/routes.rb.',
+      'For more information about routes, see the Rails guide: https://guides.rubyonrails.org/routing.html.'
+    ])
   end
 
-  sig { returns(String) }
+  sig { returns(T::Array[String]) }
   def result
-    <<~MESSAGE
-      # This is an autogenerated file for routes helper methods
-
-      # typed: strong
-      class ActionController::Base
-        extend T::Sig
-
-      #{@buffer.join("\n").indent(2)}
-      end
-    MESSAGE
+    @buffer = @parlour.rbi.split("\n")
```

Comment:
It seems this should work with just @parlour.rbi
```suggestion
    @parlour.rbi
```